### PR TITLE
Pass GCS_MAVLink object through to compass calibrators / eliminate pointless shim

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_Compass/AP_Compass_Calibration.cpp [hunk 1]
```diff
@@ -186,15 +186,17 @@ Compass::_accept_calibration_mask(uint8_t mask)
...
+bool Compass::send_mag_cal_progress(const GCS_MAVLINK& link)
 {
     uint8_t cal_mask = _get_cal_mask();
 
     for (uint8_t compass_id=0; compass_id<COMPASS_MAX_INSTANCES; compass_id++) {
         // ensure we don't try to send with no space available
-        if (!HAVE_PAYLOAD_SPACE(chan, MAG_CAL_PROGRESS)) {
-            return;
+        if (!HAVE_PAYLOAD_SPACE(link.get_chan(), MAG_CAL_PROGRESS)) {
+            // ideally we only send one progress message per call.  If
```
Comment (OXINARF) [inline libraries/AP_Compass/AP_Compass_Calibration.cpp]: ideally we would only [...]

File: libraries/AP_Compass/AP_Compass_Calibration.cpp [hunk 2]
```diff
@@ -205,27 +207,32 @@ Compass::send_mag_cal_progress(mavlink_channel_t chan)
...
+bool Compass::send_mag_cal_report(const GCS_MAVLINK& link)
 {
     uint8_t cal_mask = _get_cal_mask();
 
     for (uint8_t compass_id=0; compass_id<COMPASS_MAX_INSTANCES; compass_id++) {
         // ensure we don't try to send with no space available
-        if (!HAVE_PAYLOAD_SPACE(chan, MAG_CAL_REPORT)) {
-            return;
+        if (!HAVE_PAYLOAD_SPACE(link.get_chan(), MAG_CAL_REPORT)) {
+            // ideally we only send one progress message per call.  If
```
Comment (OXINARF) [inline libraries/AP_Compass/AP_Compass_Calibration.cpp]: Same.

File: libraries/AP_Compass/AP_Compass_Calibration.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_Compass/AP_Compass_Calibration.cpp]: Changed.

File: libraries/AP_Compass/AP_Compass_Calibration.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/AP_Compass/AP_Compass_Calibration.cpp]: Changed.

**Pull request merged**